content/master/concepts/compositions.md

```diff
@@ -1308,47 +1308,74 @@ parameters in the Composition.
-`--enable-composition-webhook-schema-validation` flag on the Crossplane pod.
 The [Crossplane Pods]({{<ref "./pods#edit-the-deployment">}}) page has
 more information on enabling Crossplane flags.
+{{< /hint >}}
+
+#### Schema aware validation modes
+
+Integrity validation errors always result in a rejection of the Composition.
+
+You can configure how Crossplane should handle both missing involved resources'
+schemas and actual schema aware validation errors.
 
 {{<hint "note" >}}
-<!-- vale write-good.TooWordy = NO -->
-Schema validation only checks the `apiVersion` and `kind` are valid. Schema
-validation doesn't validate the fields of a specific resource.
-<!-- vale write-good.TooWordy = YES -->
+In case of any involved resource schema missing, Crossplane skips schema aware
+validation altogether.
 {{< /hint >}}
 
-The default validations are still checked with schema validation enabled. 
-#### Validation modes
+The following modes are available:
 
-Schema validation supports two modes:
-* `loose` (default) - Sends an warning for each schema error and installs the 
-  Composition if the default validations pass. 
-* `strict` - Send an error for every schema validation error and rejects the
-  Composition. 
+{{< table "table table-sm table-striped" >}}
+| Mode     | Missing Schema | Schema Aware Error | Integrity Error |
+| -------- | -------------- |--------------------|-----------------|
+| `warn`   | Warning        | Warning            | Error           |
+| `loose`  | Warning        | Error              | Error           |
+| `strict` | Error          | Error              | Error           |
+{{< /table >}}
 
 Change the validation mode for a Composition with the
-{{<hover label="mode" line="5">}}crossplane.io/composition-validation-mode{{</hover>}} 
-annotation. 
+{{<hover label="mode" line="5">}}crossplane.io/composition-schema-aware-validation-mode{{</hover>}} 
+annotation.
+
+If not specified, the default mode is `warn`.
 
-For example, to enable `strict` mode checking:
+For example, to enable `loose` mode checking set the annotation value to 
+{{<hover label="mode" line="5">}}loose{{</hover>}}.
 
 ```yaml {copy-lines="none",label="mode"}
 apiVersion: apiextensions.crossplane.io/v1
 kind: Composition
 metadata:
   annotations:
-    crossplane.io/composition-validation-mode: strict
+    crossplane.io/composition-schema-aware-validation-mode: loose
   # Removed for brevity
 spec:
   # Removed for brevity
 ```
+
+{{<hint "important" >}}
+Modes also apply to Compositions defined by Configuration packages, look at
+Crossplane's logs for warnings in case of issues. The default is just to warn,
+so unless the Configuration explicitly set a different mode, Crossplane creates
+the Composition even in case of errors, but you should still check warnings as
+they can surface bugs in the Configuration or in the schema aware validation
+logic, and either the author of the Configuration or the Crossplane team should
+fix it, respectively.
```

Comment:
```suggestion
Validation modes also apply to Compositions defined by Configuration packages.

View the Crossplane pod's logs with `kubectl logs` to see warnings or errors. 

By default Composition validation only issues warnings unless the 
Configuration explicitly sets a different mode. 

Crossplane creates the Composition from a Configuration package even if there 
are errors.
```

Reply:
Splitting things up this way make it slightly incorrect.

You have to view logs to see warnings only for compositions applied by Crossplane itself, splitting it up makes it look like for all warnings you have to look at the logs, which is false.

The default mode is to only emit warnings for schema aware validation errors, not for all kinds of validation, and unless the Composition defined in the Configuration set a different mode, the default will apply. So, compositions will be successfully applied, but warnings may be printed in the logs. While from your sentence it looks like when creating compositions for packages we always ignore errors.

Reply:
Updated copy.

Reply:
`Unlike applying a Composition, Crossplane doesn’t reject errors in a Composition inside of a Configuration.` is not correct, Crossplane doesn't distinguish between Compositions coming from a user and the ones coming from a Configuration, they all hit the validating webhook and are all validated according to the same above logic, so if the mode set is something else than `warn`, a Composition coming from a Configuration could result in an error.

Reply:
Sorry, I feel like I'm going in a circle here, I apologize.

So if I apply a configuration that has a Composition with `crossplane.io/composition-schema-aware-validation-mode: strict` what happens?

Does it apply the Configuration, reject the Composition and the package doesn't become `HEALTHY`?

Reply:
Yes, exactly.

Reply:
one more rewrite. Let me know if this works.

Reply:
Not really 😅
A more correct, less vale-pleasing version would be:
```
Crossplane applies schema aware validation to Compositions defined by Configuration packages as well.

Depending on the validation mode set on the Composition, in case of schema aware validation issues,
Crossplane will log any warning to  its pod stdout or, if the Composition is fully rejected, mark the parent Configuration as unhealthy and show the error as condition's status.
```

However what I cared about is letting people know to look at logs to see warning and report them back if needed, they will notice if a composition is unhealthy no need to tell that.